src/core/manager-serialize.c

```diff
@@ -537,5 +554,15 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
                 }
         }
 
+        if (vl_server) {
+                assert(!m->varlink_server);
+
+                r = varlink_server_attach_event(vl_server, m->event, SD_EVENT_PRIORITY_NORMAL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to attach varlink connection to event loop: %m");
+
+                m->varlink_server = TAKE_PTR(vl_server);
+        }
```

Comment:
hmm, what's the rationale for doing this late here? why not doing all this already in the loop above, if `m->varlink_server` is not set?

or in other words: whenever we see `vl-ss-address=`, then: 

1. allocate a varlink server object if there is none yet and add it to the event loop
2. add the deserialized fd to it

and that's it.

Reply:
It was mostly done out of laziness of not wanting to refactor `varlink_server_attach_event()`. `varlink_server_attach_event()` sets up varlink server event loop and then loops over all of the sockets in its list to add those connect callbacks. I guess it makes more logical sense to add it as we see each socket address